### ISSUE TITLE
[clang-tidy] `modernize-use-emplace` doesn't catch this example

### ISSUE THREAD

**Issue by lrvideckis** [llvm/llvm-project#57392]
`a.cpp`:
```cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
	vector<pair<int, int>> a;
	a.push_back(make_pair(1, 2)); //caught

	vector<vector<pair<int, int>>> b(1);
	b[0].push_back(make_pair(1, 2)); //not caught

	return 0;
}
```

```console
$ clang-tidy -config="{Checks: 'modernize-use-emplace'}" a.cpp
a.cpp:6:4: warning: use emplace_back instead of push_back [modernize-use-emplace]
        a.push_back(make_pair(1, 2)); //caught
          ^~~~~~~~~~~~~~~~~~~     ~
          emplace_back
```

```console
$ clang-tidy --version
LLVM (http://llvm.org/):
  LLVM version 14.0.6
  Optimized build.
  Default target: x86_64-pc-linux-gnu
  Host CPU: skylake
```
[clang-tidy wiki](https://clang.llvm.org/extra/clang-tidy/checks/modernize/use-emplace.html)

**Comment by firewave** [comment]
Looks similar to #56721.

**Comment by firewave** [comment]
This is being detected with the latest clang-tidy version.

**Comment by firewave** [comment]
This is fixed with clang-tidy 16.
